docs/conf.py

```diff
@@ -0,0 +1,143 @@
+autodoc_mock_imports = ['h5py'
+                        'mibidata',
+                        'numpy',
+                        'matplotlib',
+                        'pandas',
+                        'skimage',
+                        'sklearn',
+                        'scipy',
+                        'seaborn',
+                        'statsmodels',
+                        'tables',
+                        'umap',
+                        'xarray']
+
+sys.modules['matplotlib.pyplot'] = mock.Mock()
+
+# prefix each section label with the name of the document it is in, followed by a colon
+# autosection_label_prefix_document = True
+autosectionlabel_prefix_document = True
+
+# what role to use for text marked up like `...`, for example this will allow you to parse `filter  ` as a cross-reference to Python function 'filter'
+default_role = 'py:obj'
+
+# use recommonmark's CommonMarkParser to parse markdown
+# might be unnecessary with m2r2 but we'll see
+source_parsers = {'.md': 'recommonmark.parser.CommonMarkParser'}
+
+# which types of file extensions we want to support
+# need .rst for index.rst, and also .md for Markdown
+source_suffix = ['.rst', '.md']
+
+# the path to the 'master' document, which in our case, is just index.rst
+# not really needed because this is the default value, but still useful to know
+master_doc = 'index'
+
+# the language we are writing the documentation in
+language = 'en'
+
+# Add any paths that contain templates here, relative to this directory.
+templates_path = ['_templates']
+
+# don't allow nbsphinx to run notebooks
+nbsphinx_execute = 'never'
+
+# List of patterns, relative to source directory, that match files and
+# directories to ignore when looking for source files.
+# This pattern also affects html_static_path and html_extra_path.
+exclude_patterns = ['_build', 'Thumbs.db', '.DS_Store', '**.ipynb_checkpoints']
+
+# custom 'stuff' we want to ignore in nitpicky mode
+# currently empty, I don't think we'll ever run in this
+# but if we do we might consider adding
+nitpick_ignore = []
+
+# -- Options for HTML output -------------------------------------------------
+
+# The theme to use for HTML and HTML Help pages.  See the documentation for
+# a list of builtin themes.
+# html_theme = 'alabaster'
+html_theme = 'sphinx_rtd_theme'
+
+# Add any paths that contain custom static files (such as style sheets) here,
+# relative to this directory. They are copied after the builtin static files,
+# so a file named "default.css" will overwrite the builtin "default.css".
+html_static_path = ['_static']
+
+# -- Options for Intersphinx config -------------------------------------------------
+
+# intersphinx mapping, for when there is a cross-reference that has no matching target
+# in the current documentation
+intersphinx_mapping = {
+    'python': ('https://docs.python.org/3.6', None),
+    'numpy': ('https://numpy.org/doc/stable', None),
+    'matplotlib': ('https://matplotlib.org/3.2.1', None)
+}
```

Comment:
I think this also adds links to types in docstrings.  That being said, we should find the standard way to name numpy arrays, pandas dataframs, xarrays, etc so they can be linked correctly.

Reply:
We can definitely address this in a future PR.